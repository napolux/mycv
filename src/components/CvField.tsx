import React from 'react';

interface ICvFieldProps {
  cvFieldName: string;
  cvFieldValue: string;
}

export default class CvField extends React.Component<ICvFieldProps> {
  render() {
    return (
      <section key={'cvfield_' + this.props.cvFieldName}>
        <div>
          <p>
            <strong>{this.mapField(this.props.cvFieldName)}</strong>: {this.mapValue(this.props.cvFieldName, this.props.cvFieldValue)}
          </p>
        </div>
      </section>
    );
  }

  mapField(s: string): string { 
    switch(s) {
      case 'dob':
        return 'Date of birth';
    }
    return s.charAt(0).toUpperCase() + s.slice(1);
  }

  mapValue(k: any, v: any): string {
    if(k === 'dob') {
      return new Date(v).toLocaleDateString('en-US');
    }
    return v;
  }
}