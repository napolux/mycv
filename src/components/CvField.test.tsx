import React from 'react';
import CvField from './CvField';
import renderer from 'react-test-renderer';

it('renders correctly the name field', () => {
  const tree = renderer
    .create(<CvField cvFieldName="name" cvFieldValue="Mario Rossi" edit={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly the date field', () => {
  const tree = renderer
    .create(<CvField cvFieldName="dob" cvFieldValue="2012-12-12" edit={false} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without edit prop', () => {
  const tree = renderer
    .create(<CvField cvFieldName="dob" cvFieldValue="2012-12-12" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});