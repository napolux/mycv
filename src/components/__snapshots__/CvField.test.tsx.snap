// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly the date field 1`] = `
<section>
  <div>
    <p>
      <strong>
        Date of birth
      </strong>
      : 
      12/12/2012
    </p>
  </div>
</section>
`;

exports[`renders correctly the name field 1`] = `
<section>
  <div>
    <p>
      <strong>
        Name
      </strong>
      : 
      Mario Rossi
    </p>
  </div>
</section>
`;
